includes/class-edd-db-customers.php

```diff
@@ -257,6 +259,50 @@ public function decrement_stats( $customer_id = 0, $amount = 0.00 ) {
 	}
 
 	/**
+	 * Updates the email address of a customer record when the email on a user is updated
+	 *
+	 * @access  public
+	 * @since   2.4
+	*/
+	public function update_customer_email_on_user_update( $user_id = 0, $old_user_data ) {
+
+		$customer = new EDD_Customer( $user_id, true );
+
+		if( ! $customer ) {
+			return false;
+		}
+
+		$user = get_userdata( $user_id );
+
+		if( $user->user_email !== $customer->email ) {
+
+			if( ! $this->get_customer_by( 'email', $user->user_email ) ) {
+
+				$success = $this->update( $customer->id, array( 'email' => $user->user_email ) );
+				
+				if( $success ) {
+					// Update some payment meta if we need to
+					$payments_array = explode( ',', $customer->payment_ids );
+
+					if( ! empty( $payments_array ) ) {
+
+						foreach ( $payments_array as $payment_id ) {
+
+							edd_update_payment_meta( $payment_id, 'email', $user->user_email );
+
+						}
+						
+					}
+
+				}
+
+			}
+
+		}
+
```

Comment:
@pippinsplugins did you test specifically PDF Vouchers? That extension manually generates an old style download url that it uses to issue the pdf vouchers